crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -83,69 +215,105 @@ impl<'db> UseDefMapBuilder<'db> {
         symbol: ScopedSymbolId,
         definition: Definition<'db>,
     ) {
+        // We have a new definition of a symbol; this replaces any previous definitions in this
+        // path.
         let def_idx = self.all_definitions.len();
         self.all_definitions.push(definition);
         self.definitions_by_symbol[symbol] = Definitions {
             #[allow(clippy::range_plus_one)]
-            definitions: def_idx..(def_idx + 1),
+            definitions_range: def_idx..(def_idx + 1),
             may_be_unbound: false,
         };
     }
 
     pub(super) fn record_use(&mut self, symbol: ScopedSymbolId, use_id: ScopedUseId) {
+        // We have a use of a symbol; clone the currently visible definitions for that symbol, and
+        // record them as the visible definitions for this use.
         let new_use = self
             .definitions_by_use
             .push(self.definitions_by_symbol[symbol].clone());
         debug_assert_eq!(use_id, new_use);
     }
 
+    /// Take a snapshot of the current visible-symbols state.
     pub(super) fn snapshot(&self) -> FlowSnapshot {
         FlowSnapshot {
             definitions_by_symbol: self.definitions_by_symbol.clone(),
         }
     }
 
+    /// Reset the current builder visible-definitions state to the given snapshot state.
     pub(super) fn set(&mut self, state: &FlowSnapshot) {
```

Comment:
I would call this `restore` or `rewind` rather than `set` to clarify that we're returning to a previous state. 

```suggestion
    pub(super) fn restore(mut self, state: &FlowSnapshot) {
```